client/cli_main.py

```diff
@@ -163,18 +174,18 @@ def main():
                         print()
                         title = input("Please choose a title for your Thought: \n\n>>TITLE: ")                        
                         message = input("What would you like to post? : \n\nMESSAGE>>: ")
-                        sym_key, enc_mess = encrypt_message_symmetrical(message)
+                        enc_mess = bytes(encrypt_message_symmetrical(message))
                         print()
                         
-                        post_thought(server_url, username, title, enc_mess)                         
+                        post_thought(server_url, username, title, enc_mess)
                                      
                         print("Message uploaded successfully!")
                                             
                     elif sub_choice == "3":
                         get_user_friends(server_url)
                         print()
                         
-                        base_64_encr_sym_key = None
+                        base_64_encr_sym_key = str()
```

Comment:
Not sure this is going to work if you change it to a string already. The problem here is that if you don't enter a friends username for example when you want to read messages it returns and empty variable for the key above or a dictionary. The check is to only continue if the returned value is a string.

Reply:
That's why it is an empty string. I'll do some more testing to test the case.

Reply:
This is resolved. It works flawlessly now!